Kconfig.thingset

```diff
@@ -27,6 +27,22 @@ config THINGSET_IMMUTABLE_OBJECTS
 
 	  Possible future improvement: Store only subset information in RAM and the rest in ROM.
 
+config THINGSET_OBJECT_LOOKUP_MAP
+	bool "Enable hashmap for object lookup"
+	default n
+	help
+	  Enable to speed up performance of object lookup.
+
+if THINGSET_OBJECT_LOOKUP_MAP
+
+config THINGSET_OBJECT_LOOKUP_BUCKETS
+	int "Object lookup hashmap buckets"
+	default 8
+	help
+	  Number of buckets for object lookup
```

Comment:
Should we mention that this should ideally be a power-of-two number for the modulo operations to be efficient?